lib/ResolverFactory.js

```diff
@@ -45,23 +72,33 @@ exports.createResolver = function(options) {
 	// changes the original options.plugins array, causing duplicate plugins
 	const plugins = (options.plugins && options.plugins.slice()) || [];
 
-	// A list of main fields in description files
-	let mainFields = options.mainFields || ["main"];
-
 	// A list of alias fields in description files
 	const aliasFields = options.aliasFields || [];
 
 	// A list of main files in directories
 	const mainFiles = options.mainFiles || ["index"];
 
-	// A list of extensions which should be tried for files
-	let extensions = options.extensions || [".js", ".json", ".node"];
+	/**
+	 * A list of extensions which should be tried for files
+	 * @type {Array<string>}
+	 */
+	const extensions = options.extensions
+		? [].concat(/** @type {Array<never>} */ (options.extensions))
+		: [".js", ".json", ".node"];
 
 	// Enforce that a extension from extensions must be used
 	const enforceExtension = options.enforceExtension || false;
 
-	// A list of module alias configurations or an object which maps key to value
-	let alias = options.alias || [];
+	/**
+	 * A list of module alias configurations or an object which maps key to value
+	 * @type {Array<AliasOptionEntry>}
+	 */
+	const alias =
```

Comment:
Looks like `{alias?: AliasOptionNewRequest, name?: string, onlyModule?: boolean}` is not correct type, but I have defined as webpack^4.0.0 defines. Seems like config below will be broken:
```
module.exports = {
  resolve: {
     alias: [{alias: 'blabla'}]
  }
};
```

https://github.com/webpack/webpack/blob/master/schemas/WebpackOptions.json#L1076 - does not define all fields as required

Reply:
https://github.com/webpack/webpack/issues/9404

Reply:
fix type